### DISCUSSION TITLE
Android usage

### DISCUSSION THREAD

**Discussion by ellykits** [evrete/evrete#46] [Q&A] [answered]
Is it possible to use this library in an android application?

**Comment by andbi** [comment, marked as answer]
It is possible for `minSdk 26` as long as you use Java predicates as rule conditions. Afaik, Android totally lacks instrumentation API, so the library will fail to compile literal conditions like `"$obj.id > 0"`. Please also make sure you use the latest (3.x.x) library versions with dropped support for Java SecurityManager.
